deploy/examples/nvmeof-test.yaml

```diff
@@ -0,0 +1,69 @@
+# This example is for Ceph v19 and above only
+# The CephBlockPool must be created before the CephNVMeOFGateway
+#
+# Note: The operator will automatically create the following resources:
```

Comment:
Please create a document that describes how the user can use this feature. Some thoughts:
- Create the doc: Documentation/Storage-Configuration/Block-Storage-RBD/nvme-of.md
- The content will be similar as block-storage.md, to describe the scenario, show a simple yaml example, and describe how the user can enable the nvmeof csi driver, and do all the steps to configure the scenario. 

With this document, reviewers can even better understand the end-to-end flow to better review the feature, and then the feature will be documented for users to try out.